src/lib/Webhook.coffee

```diff
@@ -0,0 +1,61 @@
+request = require "requestretry"
+
+DEFAULT_TIMEOUT = 10 * 1000
+DEFAULT_MAX_ATTEMPTS = 3
+
+class Webhook
+
+  constructor: (options = {}) ->
+    @options = options
+    @timeout = DEFAULT_TIMEOUT
+    @maxAttempts = DEFAULT_MAX_ATTEMPTS
+
+  @detectEmoji: (emoji) ->
+    obj = {}
+
+    unless emoji
+      obj.key = "icon_emoji"
+      obj.val = ""
+      return obj
+
+    obj.key = if emoji.match(/^http/) then "icon_url" else "icon_emoji"
+    obj.val = emoji
+    return obj
+
+  respond: (options, callback) ->
+
+    if typeof options is "object"
```

Comment:
Note `typeof null` returns "object", if someone passed that in we'd NPE on the next line.